### PR TITLE
feature(pu): add game_2048_env and configs

### PR DESCRIPTION
**Pull request by puyuan1996**
- add game_2048_env and configs

### CODE REVIEW

File: agent/config/muzero_2048_config.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+from easydict import EasyDict
```
Comment (PaParaZz1) [inline agent/config/muzero_2048_config.py:1]: remove origin `test_env.py`

File: agent/config/muzero_2048_config.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
```
Comment (PaParaZz1) [inline agent/config/muzero_2048_config.py:2]: update requirements.txt

File: agent/config/muzero_2048_config.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+max_env_step = int(5e6)
+reanalyze_ratio = 0.
+num_of_possible_chance_tile = 2
+chance_space_size = 16 * num_of_possible_chance_tile
+# ==============================================================
+# end of the most frequently changed config specified by the user
+# ==============================================================
+
+atari_muzero_config = dict(
+    exp_name=f'data_muzero/game_2048_npct-{num_of_possible_chance_tile}_muzero_ns{num_simulations}_upc{update_per_collect}_rer{reanalyze_ratio}_bs{batch_size}_sslw2_seed0',
```
Comment (PaParaZz1) [inline agent/config/muzero_2048_config.py]: simplify name here, this repo does not need too many hyper-parameters adjustment.

File: agent/config/muzero_2048_config.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+from easydict import EasyDict
+
+
```
Comment (PaParaZz1) [inline agent/config/muzero_2048_config.py:3]: update readme

File: agent/config/muzero_2048_config.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+reanalyze_ratio = 0.
+num_of_possible_chance_tile = 2
+chance_space_size = 16 * num_of_possible_chance_tile
+# ==============================================================
+# end of the most frequently changed config specified by the user
+# ==============================================================
+
+atari_muzero_config = dict(
+    exp_name=f'data_muzero/game_2048_npct-{num_of_possible_chance_tile}_muzero_ns{num_simulations}_upc{update_per_collect}_rer{reanalyze_ratio}_bs{batch_size}_sslw2_seed0',
+    env=dict(
```
Comment (PaParaZz1) [inline agent/config/muzero_2048_config.py:25]: add a basic training curve

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: agent/config/muzero_2048_config.py (hunk 5 above)
Comment (puyuan1996) [reply agent/config/muzero_2048_config.py:25]: done

File: agent/envs/game_2048_env.py [hunk 6]
```diff
@@ -0,0 +1,794 @@
...
+      - step(action):
+          Advances the game by one step based on the provided action. It returns the new observation, reward, game status,
+          and additional information.
+      - render(mode='human'):
+          Renders the current state of the game for visualization purposes.
+    MDP Definition:
+      - Observation Space:
+          The observation space is a 4x4 grid representing the game board. Each cell in the grid can contain a number from
+          0 to 2048. The observation can be in different formats based on the 'obs_type' parameter in the environment configuration.
+          - If 'obs_type' is set to 'encode_observation' (default):
```
Comment (zjowowen) [inline agent/envs/game_2048_env.py]: 'encode_observation'->'raw_encoded_board'?

Comment (puyuan1996) [reply agent/envs/game_2048_env.py]: done

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**